platform/viewer/src/connectedComponents/ConnectedStudyBrowser.js

```diff
@@ -28,13 +31,55 @@ const mapStateToProps = (state, ownProps) => {
   });
 
   return {
+    activeViewportIndex: state.viewports.activeViewportIndex,
     studies: studiesWithLoadingData,
   };
 };
 
+const mapDispatchToProps = dispatch => {
+  return {
+    setViewportSpecificData: (viewportIndex, data) => {
+      dispatch(setViewportSpecificData(viewportIndex, data));
+    },
+  };
+};
+
+const mergeProps = (propsFromState, propsFromDispatch, ownProps) => {
+  const findDisplaySet = (studies, studyInstanceUid, displaySetInstanceUid) => {
+    const study = studies.find(study => {
+      return study.studyInstanceUid === studyInstanceUid;
+    });
+
+    if (!study) {
+      return;
+    }
+
+    return study.displaySets.find(displaySet => {
+      return displaySet.displaySetInstanceUid === displaySetInstanceUid;
+    });
+  };
+
+  return {
+    ...propsFromState,
+    onThumbnailClick: data => {
+      const displaySet = findDisplaySet(
+        ownProps.studies,
+        data.studyInstanceUid,
+        data.displaySetInstanceUid
+      );
+
+      propsFromDispatch.setViewportSpecificData(
+        propsFromState.activeViewportIndex,
+        displaySet
+      );
```

Comment:
Do we have a way to change the active displaySet without passing it in it's entirety? Ideally, we should be able to just set the stack by index, study id, displayset id, etc.

It feels like overkill to surface this amount of information in a click event.

Reply:
That was actually a very good callout @dannyrb thanks for that
It called out my attention on other reducers and also to test how many data do I need to update the specific data on the viewport. Turns out, only the displaySet works fine.